src/docs/process-binding.md

```diff
@@ -0,0 +1,87 @@
+# The Process of Binding
+
+In theory, the process of binding a Swift binary into C# should be as simple as:
+
+1. Generate and consume the abi.json file
+2. Extract and demangle the symbols from the binary
+3. Iterate over every type and function and gnerate C# and/or supporting Swift code
+4. Generate type datatbase entries
```

Comment:
```suggestion
4. Generate type database entries
```

Reply:
How do you envision the type database structure and usage? In the context of this PR, are the handlers supposed to populate the database as they process the entity?

Reply:
I envision there being likely two type databases. The first is a bind-time database which would need maximal information about the types: Swift module, Swift name, C# namespace, C# name, entity type, blitability. But this is something that will be in its own document. The database would be populated with new entries after the ability.json file has been parsed but before binding. Older entries can be read in before the abi.json file is read. The second database is run-time. Anything we can do to minimize the size and start-up overhead. For the most part, the run-time database is to support the generic programming model, but again, this would be it's own document.